apps/testing/29-real-life-application/src/app/list/ticket.store.spec.ts

```diff
@@ -1,28 +1,137 @@
+import { TestBed } from '@angular/core/testing';
+import { provideComponentStore } from '@ngrx/component-store';
+import { createMockWithValues } from '@testing-library/angular/jest-utils';
+import { lastValueFrom, of, take, throwError } from 'rxjs';
+import { BackendService } from '../backend.service';
+import { TicketStore } from './ticket.store';
+
+const USERS = [
+  { id: 1, name: 'titi' },
+  { id: 2, name: 'george' },
+];
+const TICKETS = [
+  {
+    id: 0,
+    description: 'Install a monitor arm',
+    assigneeId: 1,
+    completed: false,
+  },
+  {
+    id: 1,
+    description: 'Coucou',
+    assigneeId: 1,
+    completed: false,
+  },
+];
+
 describe('TicketStore', () => {
   describe('When init', () => {
+    function setup({ userSuccess = true }: { userSuccess?: boolean } = {}) {
+      const mockService = createMockWithValues(BackendService, {
+        users: jest.fn(),
+        tickets: jest.fn(),
+      });
+      if (userSuccess) {
+        mockService.users.mockReturnValueOnce(of(USERS));
+      } else {
+        mockService.users.mockReturnValueOnce(throwError(() => 'oops'));
+      }
+      mockService.tickets.mockReturnValueOnce(of(TICKETS));
+
+      TestBed.configureTestingModule({
+        providers: [
+          provideComponentStore(TicketStore),
+          {
+            provide: BackendService,
+            useValue: mockService,
+          },
+        ],
+      });
```

Comment:
why do you prefer to use Testbed instead of the render function with an emptu component ?

Reply:
Semantically, it feels weird to me to use the word "render" with something that has no visual component.